ftl/python/builder.py

```diff
@@ -46,19 +66,39 @@ def Build(self):
             interpreter_builder.BuildLayer()
             lyr_imgs.append(interpreter_builder.GetImage())
 
-            # build package layers
-            req_txt_builder = package_builder.RequirementsLayerBuilder(
-                ctx=self._ctx,
-                descriptor_files=self._descriptor_files,
-                pkg_dir=None,
-                wheel_dir=self._wheel_dir,
-                venv_dir=self._venv_dir,
-                pip_cmd=self._pip_cmd,
-                venv_cmd=self._venv_cmd,
-                dep_img_lyr=interpreter_builder,
-                cache=cache)
-            req_txt_builder.BuildLayer()
-            lyr_imgs.append(req_txt_builder.GetImage())
+            if self._is_phase2:
```

Comment:
another design nit: since the logic between phase 1 and phase 2 is so different and we check to see which one we're doing so many times, I was consider splitting them up into two separate builders for each language. then in the top level builder we can just choose the one we want there and never have to check which phase we're in again. don't necessarily have to do it here but I think we should at some point

Reply:
I agree we should split it out and refactor into separate builders for phase 1 and phase 2.  Typically the only difference though is something like:
phase 1: pass depfile to builder, install all packages
phase 2: iterate over depfile, install single package